apps/web/src/services/iam/components/UserGroupManagementTabNotificationChannel.vue

```diff
@@ -59,37 +68,102 @@ const tableState = reactive({
 const handleSelect = async (index: number[]) => {
     userGroupPageState.userGroupChannels.selectedIndices = index;
 };
 
-const handleUpdateModal = (modalType: string) => {
-    switch (modalType) {
-    case 'add':
+const handleUpdateModal = async (modalType: string) => {
+    if (modalType === 'create') {
         userGroupPageStore.updateModalSettings({
             type: USER_GROUP_MODAL_TYPE.CREATE_NOTIFICATIONS_FIRST,
             title: i18n.t('IAM.USER_GROUP.MODAL.CREATE_CHANNEL.TITLE'),
             themeColor: 'primary1',
         });
-        break;
-    //     TODO: Update with stored values
-    case 'edit':
+    } else if (modalType === 'edit') {
+        const result = await fetchGetUserGroupChannel({
+            channel_id: userGroupPageGetters.selectedUserGroupChannel[0].channel_id,
+        });
+        if (result) {
+            notificationChannelCreateFormStore.setSelectedProtocol(result.protocol_id);
+            notificationChannelCreateFormStore.setChannelName(result.name);
+            if (result.data) {
+                if (result.data.USER && result.data.USER.length > 0) {
+                    notificationChannelCreateFormStore.setUserInfo({
+                        type: 'USER',
+                        value: result.data.USER,
+                    });
+                } else if (result.data.USER_GROUP && result.data.USER_GROUP.length > 0) {
+                    notificationChannelCreateFormStore.setUserInfo({
+                        type: 'USER_GROUP',
+                        value: result.data.USER_GROUP,
+                    });
+                }
+            }
+            if (result.schedule.SCHEDULE_TYPE === 'ALL_DAY') {
+                notificationChannelCreateFormStore.setScheduleInfo({
+                    days: ['MON', 'TUE', 'WED', 'Thu', 'Fri', 'Sat', 'Sun'],
+                    start: result.schedule.MON.start,
+                    end: result.schedule.MON.end,
+                    type: 'ALL_DAY',
+                });
+            } else if (result.schedule.SCHEDULE_TYPE === 'WEEK_DAY') {
+                notificationChannelCreateFormStore.setScheduleInfo({
+                    days: ['MON', 'TUE', 'WED', 'Thu', 'Fri'],
+                    start: result.schedule.MON.start,
+                    end: result.schedule.MON.end,
+                    type: 'WEEK_DAY',
+                });
+            } else if (result.schedule.SCHEDULE_TYPE === 'CUSTOM') {
+                notificationChannelCreateFormStore.setScheduleInfo({
+                    days: Object.values(result.schedule).map((sc, idx) => {
+                        if (typeof sc === 'object' && sc.is_scheduled) {
+                            return Object.keys(result.schedule)[idx];
+                        }
+                        return null;
+                    }),
+                    start: result.schedule.MON.start,
+                    end: result.schedule.MON.end,
+                    type: 'CUSTOM',
+                });
```

Comment:
🤔: Can't we simplify it by creating a function? I don't think manually entering the date is a good approach!